### ISSUE TITLE
Make threaded isodoc execution optional

### ISSUE THREAD

**Issue by opoudjis** [metanorma/metanorma#230]
Follows from https://github.com/metanorma/metanorma/issues/220

Add an option to disable this, through an environment variable, for Ruby-packer environment.

@andrew2net has already done the same for relaton: https://github.com/relaton/relaton/commit/7663993775fdbb500cabfc58b03eda295e933e02

**Comment by opoudjis** [comment]
Also need to mutex all file accesses within isodoc.

**Comment by opoudjis** [comment]
Will reuse `RELATON_FETCH_PARALLEL` variable for metanorma as well.

**Comment by ronaldtse** [comment]
@opoudjis please use a Metanorma specific environment variable. Thanks.

**Comment by opoudjis** [comment]
Done: `METANORMA_FETCH_PARALLEL`

**Comment by ronaldtse** [comment]
@opoudjis Metanorma isn't "fetching" in parallel though. How about `METANORMA_PARALLEL={number}`?

**Comment by opoudjis** [comment]
OH FFS. FINE.

**Comment by opoudjis** [comment]
Created Metanorma::FileHelper singleton class with Mutex to do threadsafe file reads (and if needed writes).
